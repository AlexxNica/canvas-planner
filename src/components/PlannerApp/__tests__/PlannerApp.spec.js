import React from 'react';
import { shallow } from 'enzyme';
import 'instructure-ui/lib/themes/canvas';
import { PlannerApp } from '../index';

const getDefaultValues = (overrides) => (
  Object.assign({}, {
    dayKeys: ["2017-04-24", "2017-04-25", "2017-04-26"],
    days: {
      "2017-04-24": [{}],
      "2017-04-25": [{}],
      "2017-04-26": [{}]
    },
    timeZone: "en"
  }, overrides)
);

it('renders base component using dayKeys', () => {
  const wrapper = shallow(
    <PlannerApp {...getDefaultValues()} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('shows only the loading component when the isLoading prop is true', () => {
  const wrapper = shallow(
    <PlannerApp
      {...getDefaultValues()}
      isLoading={true}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
