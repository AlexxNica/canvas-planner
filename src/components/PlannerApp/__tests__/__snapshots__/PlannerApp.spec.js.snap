// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders base component using dayKeys 1`] = `
<div
  className="PlannerApp"
>
  <Day
    day="2017-04-24"
    itemsForDay={
      Array [
        Object {},
      ]
    }
    timeZone="en"
  />
  <Day
    day="2017-04-25"
    itemsForDay={
      Array [
        Object {},
      ]
    }
    timeZone="en"
  />
  <Day
    day="2017-04-26"
    itemsForDay={
      Array [
        Object {},
      ]
    }
    timeZone="en"
  />
</div>
`;

exports[`shows only the loading component when the isLoading prop is true 1`] = `
<div
  className="PlannerApp"
>
  <Container
    display="block"
    padding="xxLarge medium"
    textAlign="center"
  >
    <Spinner
      size="medium"
      title="Loading planner items"
      variant="default"
    />
  </Container>
</div>
`;
